src/method.rs

```diff
@@ -156,6 +159,24 @@ impl Default for Method {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<http_types::Method> for Method {
+    fn from(method: http_types::Method) -> Method {
+        method.as_ref().parse()
```

Comment:
These `Method` conversions can probably skip `parse` in most cases, and just match on the common variants, so the cost should be minuscule.